File: mergeArray.java
/* Time complexity: O(n), where n is the size of the smaller array
space complexity: O(1)

1. Given that array1 is the bigger array, start comparing the elements from last position
and keep appending it in the later positions of the bigger array.
2. Keep doing this until you iterate through the entire array 1 or array2
3. In case any elements are left to be sorted, in array2, just append it in front of array1.

*/
class Solution{
    public void merge(int[] nums1, int m, int[] nums2, int n){
        int pt1 = m-1;
        int pt2 = n-1;
        int index = m+n-1;
        while(pt1 >= 0 && pt2 >= 0){
            if(nums1[pt1] > num2[pt2]){
                nums1[index] = num1[pt1];
                pt1--;
            }
            else{                
                nums1[index] = num2[pt2];
                pt2--;            
            }
            index--;
        }
        while(pt2 >= 0){
            nums1[index] = nums2[pt2];
            pt2--;
            index--;
        }
    }
}

File: Mergearray.py
class Solution:
    def merge(self, nums1, m: int, nums2, n: int) -> None:
        p1=m-1
        p2=n-1
        idx=m+n-1
        while(p1>=0 and p2>=0):
            if(nums1[p1]>=nums2[p2]):
                nums1[idx]=nums1[p1]
                p1-=1

            else:
                nums1[idx]=nums2[p2]
                p2-=1
            idx-=1
        while(p2>=0):
            nums1[idx]=nums2[p2]
            p2-=1
            idx-=1


nums1 = [4, 5, 6, 0, 0, 0]
nums2=[1,2,3]
m=3
n=3
obj=Solution()
obj.merge(nums1,m,nums2,n)
print(nums1)